src/regex/engine.cr

```diff
@@ -1,4 +1,11 @@
-require "./pcre"
+{% if flag?(:force_pcre2) || (!flag?(:force_pcre) && `hash pkg-config 2> /dev/null && pkg-config --silence-errors --modversion libpcre2-8 || printf %s false` != "false") %}
```

Comment:
[`-Duse_libiconv`](https://github.com/crystal-lang/crystal/issues/11882) does a similar thing as `-Dforce_pcre` and [`-Dforce_iouring`](https://github.com/crystal-lang/crystal/pull/10768), so I think we should unify the verbs here.

Later I think we should extend this `Engine` idiom to iconv and the event loop.

Reply:
I suppose `use_x` sounds a bit better. It's not as strong as "force".